pkg/eventmonitor/consumers/process.go

```diff
@@ -42,32 +49,46 @@ type event struct {
 	pid       uint32
 }
 
-// ProcessConsumer should implement the EventConsumerHandler and EventConsumer interfaces
-var _ eventmonitor.EventConsumerHandler = &ProcessConsumer{}
-var _ eventmonitor.EventConsumer = &ProcessConsumer{}
+// ProcessConsumerEventTypes represents the types of events that the ProcessConsumer can handle, a subset
+// of the model.EventType
+type ProcessConsumerEventTypes model.EventType
 
-// ProcessConsumerOptions holds the options that can be passed to NewProcessConsumer
-type ProcessConsumerOptions struct {
-	// ID is the ID of the consumer that will be used to identify it with the event data stream
-	ID string
+const (
+	// ExecEventType represents process open events
+	ExecEventType ProcessConsumerEventTypes = ProcessConsumerEventTypes(model.ExecEventType)
 
-	// ChanSize is the size of the channel that the event monitor will use to send events to this handler
-	ChanSize int
+	// ExitEventType represents process exit events
+	ExitEventType ProcessConsumerEventTypes = ProcessConsumerEventTypes(model.ExitEventType)
 
-	// ListenToForkEvents indicates if this consumer should listen to and handle fork events
-	ListenToForkEvents bool
-}
+	// ForkEventType represents process fork events
+	ForkEventType ProcessConsumerEventTypes = ProcessConsumerEventTypes(model.ForkEventType)
+)
+
+// ProcessConsumer should implement the EventConsumerHandler and EventConsumer interfaces
+var _ eventmonitor.EventConsumerHandler = &ProcessConsumer{}
+var _ eventmonitor.EventConsumer = &ProcessConsumer{}
 
 // NewProcessConsumer creates a new ProcessConsumer, registering itself with the
 // given event monitor. This function should be called with the EventMonitor
 // instance created in cmd/system-probe/modules/eventmonitor.go:createEventMonitorModule.
 // For tests, use consumers/testutil.NewTestProcessConsumer, which also initializes the event stream for testing accordingly
-func NewProcessConsumer(options ProcessConsumerOptions, evm *eventmonitor.EventMonitor) (*ProcessConsumer, error) {
+func NewProcessConsumer(id string, chanSize int, eventTypes []ProcessConsumerEventTypes, evm *eventmonitor.EventMonitor) (*ProcessConsumer, error) {
 	pc := &ProcessConsumer{
-		options:       options,
-		execCallbacks: callbackMap{callbacks: make(map[*ProcessCallback]struct{})},
-		exitCallbacks: callbackMap{callbacks: make(map[*ProcessCallback]struct{})},
-		forkCallbacks: callbackMap{callbacks: make(map[*ProcessCallback]struct{})},
+		id:         id,
+		chanSize:   chanSize,
+		eventTypes: eventTypes,
+	}
+
+	if pc.isEventTypeEnabled(ExecEventType) {
+		pc.execCallbacks = newCallbackMap()
+	}
+
+	if pc.isEventTypeEnabled(ExitEventType) {
+		pc.exitCallbacks = newCallbackMap()
+	}
+
+	if pc.isEventTypeEnabled(ForkEventType) {
+		pc.forkCallbacks = newCallbackMap()
 	}
```

Comment:
don't you need to check for nil pointer everywhere you use those callback maps now? (e.g: [HandleEvent](https://github.com/DataDog/datadog-agent/pull/30559/files/26768ee76ddad7308c4418d63681ce24fe57b11f..99baa4e8a14723aa851e5b66c3cd7edcba379039#diff-42bf6324ee3c6c4e49f16275eb9ab8326a280d0abc72b8601d8393d62c7505f6R157) func)

Reply:
I thought about it, but I think it's best to not do that. The function signatures we have now don't support returning errors, so we'd have to resort to logs. In this case, if someone makes a mistake, it will be something like subscribing to events without enabling them, or the event data stream sending events of the wrong type. That is, a programming mistake that will always cause the same problem. I think in that case is better to have a panic, which will always surface in unit tests, e2e tests or in the worst case staging deployments, rather than a log that gets way too easily ignored and will cause hard-to-debug issues where consumers don't actually receive the events they're expecting to.

Reply:
Either way, I'll add a comment clearly explaining what we end up deciding.